File: docs/scripts/model/rules/endofround/DrawDouble.js
import ResultType from "../../ResultType.js";

export default class DrawDouble {

    constructor() {
        this.draw = false;
    }

    execute(r, players) {
        let index = -1;
        switch (r.getWinner()) {
            case ResultType.DRAW:
                draw = true;
                break;
            default:
                index = r.getWinner();
                let winner = players.get(index);
                let oldPoints = winner.getPoints();
                winner.setPoints(oldPoints + (draw ? 2 * r.getPoints() : r.getPoints()));
                draw = false;
                break;
        }
        return index;
    }
}

File: src/main/java/model/rules/drawround/DrawDouble.java
package model.rules.drawround;

import model.RoundResult;

public class DrawDouble implements DrawRound {

    private boolean draw = false;

    @Override
    public int execute(RoundResult r) {
        int score = r.score();
        switch (r.winner()) {
            case RoundResult.DRAW -> {
                draw = true;
            }
            default -> {
                score = draw ? 2 * r.score() : r.score();
                draw = false;
            }
        }
        return score;
    }

}
